Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -871,7 +990,7 @@ public boolean onTouch(View v, MotionEvent ev) {
 	 */
 	public void closeSession() {
 		if (prefHelper_.getSmartSession()) {
-			if (keepAlive_) {
+			if (keepAlive_ && !isBranchAppInstantiated) { //No need to check for keepAlive_ for auto session management
```

Comment:
New Session Management is also using closeSession().Only difference is that close session is called only when the application goes to the background not with each activity stop.
So executeClose() is still called when the last  activity leaves the screen